### PR TITLE
Fixes(milestone): Multiple Milestone on spam clicking.

### LINKED ISSUE #9687: [BUG] <adds multiple milestones>
### Has this bug been raised before?

- [X] I have checked "open" AND "closed" issues and this is not a duplicate

### Where did you find this bug?

while adding milestone sectio

### Version of BioDrop (for example "v1.2.3")

2.83.14

### Description

when clicked multiple times on save button in milestone section, it sends many requests which results in adding of the same milestone multiple times 

### Screenshots

Screenshot 2023-10-30 at 9 13 16 PM

### Do you want to work on this issue?

Yes

### If you want to work on this issue...

By disabling the button when the request is sent.

### PR DESCRIPTION
**Pull request by malay44**
Introduce `isDisabled` state to `manage/milestone` page for save button control.

## Fixes Issue

Closes #9687 

## Changes proposed

- Added a `isDisabled` state to disable the save button upon clicking it.

## Check List (Check all the applicable boxes) 

- [X] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [X] This PR does not contain plagiarized content.
- [X] The title of my pull request is a short description of the requested changes.

## Note to reviewers

I am having errors while running the tests in gitpod, but I have tested my changes in a self-hosted version.

### CODE REVIEW

**Comment by eddiejaoude** [comment]
Looks good 👍 

I like Sital's suggestion 👍

File: pages/account/manage/milestone/[[...data]].js [hunk 1]
```diff
@@ -274,7 +277,7 @@ export default function ManageMilestone({ BASE_URL, milestone }) {
                       DELETE
                     </Button>
                   )}
-                  <Button type="submit" primary={true}>
+                  <Button className={isDisabled && "cursor-not-allowed"} type="submit" primary={true} disabled={isDisabled}>
```
Comment (eddiejaoude) [inline pages/account/manage/milestone/[[...data]].js:280]: Please add this to the `Button` component, so it works everywhere `components/Button.js`

Comment (eddiejaoude) [reply pages/account/manage/milestone/[[...data]].js:280]: I noticed this already exists on the component, if you use the prop `disable`

**Comment by sital002** [comment]
Looks good  👍
Can you implement this in all these places?
```pages/account/manage/milestone/[[...data]].js```
```pages/account/manage/link/[[...data]].js```
```pages/account/manage/event/[[...data]].js```
```pages/account/manage/repos.js```
```pages/account/manage/profile.js```
```pages/account/manage/premium.js```

**Comment by malay44** [comment]
Screenshot 2023-11-03 at 9 23 09 AM

Navbar text is coming on top of the notification. I tried to adjust the z index of the notification, but there was no change.

**Comment by sital002** [comment]
> Navbar text is coming on top of the notification. I tried to adjust the z index of the notification, but there was no change.

There is already a existing issue related to this https://github.com/EddieHubCommunity/BioDrop/issues/9618

**Comment by eddiejaoude** [comment]
Looks good 👍  please can you resolve conflicts

**Comment by malay44** [comment]
Hello, I've addressed the merge conflicts mentioned earlier. Feel free to propose any additional modifications or optimizations. 😊

File: pages/account/manage/event/[[...data]].js [hunk 2]
```diff
@@ -349,7 +352,7 @@ export default function ManageEvent({ BASE_URL, event }) {
                       DELETE
                     </Button>
                   )}
-                  <Button type="submit" primary={true}>
+                  <Button type="submit" primary={true} disable={isDisabled}>
```
Comment (sital002) [inline pages/account/manage/event/[[...data]].js:355]:
```suggestion
                  <Button type="submit" primary={true} disabled={isDisabled}>
```
button takes ```disabled``` as props not ```disable```

File: pages/account/manage/link/[[...data]].js [hunk 3]
```diff
@@ -265,7 +268,7 @@ export default function ManageLink({ BASE_URL, username, link }) {
                       DELETE
                     </Button>
                   )}
-                  <Button type="submit" primary={true}>
+                  <Button type="submit" primary={true} disable={isDisabled}>
```
Comment (sital002) [inline pages/account/manage/link/[[...data]].js:271]:
```suggestion
                  <Button type="submit" primary={true} disabled={isDisabled}>
```

File: pages/account/manage/milestone/[[...data]].js [hunk 4]
```diff
@@ -274,7 +277,7 @@ export default function ManageMilestone({ BASE_URL, milestone }) {
                       DELETE
                     </Button>
                   )}
-                  <Button type="submit" primary={true}>
+                  <Button type="submit" primary={true} disable={isDisabled}>
```
Comment (sital002) [inline pages/account/manage/milestone/[[...data]].js:280]:
```suggestion
                  <Button type="submit" primary={true} disabled={isDisabled}>
```

File: pages/account/manage/profile.js [hunk 5]
```diff
@@ -259,7 +262,7 @@ export default function Profile({ BASE_URL, profile, fileExists }) {
                 </section>
 
                 <div className="mt-10 border-t border-primary-low-medium/30 pt-6 sm:flex sm:items-center sm:justify-between">
-                  <Button primary={true}>SAVE</Button>
+                  <Button primary={true} disable={isDisabled}>SAVE</Button>
```
Comment (sital002) [inline pages/account/manage/profile.js:265]:
```suggestion
                  <Button primary={true} disabled={isDisabled}>SAVE</Button>
```

**Review by sital002 — COMMENTED** [review]
Sorry for the late reply, I have left some inline comments.

**Review by eddiejaoude — APPROVED** [review]
Thank you 👍

**Pull request merged**